components/edit_channel_purpose_modal/edit_channel_purpose_modal.tsx

```diff
@@ -1,53 +1,49 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ChangeEvent, KeyboardEvent} from 'react';
 import {Modal} from 'react-bootstrap';
-import {FormattedMessage, injectIntl} from 'react-intl';
+import {FormattedMessage, injectIntl, IntlShape} from 'react-intl';
+
+import {Channel} from 'mattermost-redux/types/channels';
+import {ActionResult} from 'mattermost-redux/types/actions';
 
 import Constants from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 
+type Actions = {
+    patchChannel: (channelId: string, patch: Partial<Channel>) => Promise<ActionResult>;
+}
+type ServerError = {
```

Comment:
This type already exists in `package/mattermost-redux` so we don't need to redefine it here.